doc: add minutes for meeting 2 Aug 2023 (#1425)
nodejs/TSC: meetings/2023-08-02.md

  * Michael - another naive question not having read through the issue - just read a file with the options from a file as a string instead of environment variables.
  * Tobias - popularity is not necessary the bar,  non-contributors always like new features. If we had added every requested feature, Node.js would contain every single npm package in core.
    * Rich +1
  * Benjamin - do we have consensus on landing with NODE_OPTIONS support
    * Geoffrey - best to work for a few weeks to try to land together first
  * Tobias - What about environment variables that affect dependencies such as UV_THREADPOOL_SIZE? If we really add this to core, it should cover the vast majority of use cases.
  * All - worth people looking at - <https://openjs-foundation.slack.com/archives/C027PSG2PJR/p1690225162828989>
  * Moshe, .env support is really a different feature than the support for NODE_OPTIONS, configuring options for Node.js is a separate concern
  * Antoine - agree its a separate concern, don’t see point of waiting.
  * Matteo, may be an argument about not supporting anything that applies to Node.js.

>     * Geoffrey but then that is the same as .env

Reviewer: ```suggestion
    * Geoffrey but then that is the same as userland .env
```

>   * Michael but if whole point is to configure Node.jh how does it makes sense to land without NODE_OPTIONS  and other Node environment

Reviewer 2: ```suggestion
  * Michael but if whole point is to configure Node.js how does it makes sense to land without NODE_OPTIONS and other Node.js environment variables
```

>   * Michael but if whole point is to configure Node.jh how does it makes sense to land without NODE_OPTIONS  and other Node environment  

Author: ```suggestion
  * Michael but if whole point is to configure Node.jh how does it makes sense to land without NODE_OPTIONS  and other Node environment
```

  * Geoffrey, agree with Antoine but that is not necessarily the blocker for me, but block is that this should be standardized and agreed in WinterCG before this would be added. Past experience is that if not standardized, will cause issues later on. Would be fine to add them once WinterCG agrees on a standard - <https://github.com/wintercg/proposal-common-minimum-api/issues/50>
  * Matteo - have been one of the people who have been frustrated by this. Discussions have not been productive. Seems like people believe that it should be agnostic or that it needs to be aware of how/where it runs. What I like about Node.js is that you have that affinity with the file system. In the past it was not clear how we could provide the info. It’s a philosophical discussion so that’s why called vote as unlikely to get agreement. Getting into WinterCG is a separate discussion. Change does not remove anything from anybody not using it.
  * Yagiz, Benjamin + 1 to what Matteo mentioned.
  * Michael, the lack of this information has been raised as one of the blockers in moving to ESM in the latest Node.js survey.

>   * From WinterCG viewpoint, should be treated like X (missed it), we should treat things in global scope which import.meta is. Should be treated as SemVer major.  Already plan to add `import.meta.url` and `import.meta.resolve` to the minimum. That said, ok to add the bit of metadata being proposed, but ask would be some write up to WinterCG to get input to avoid other runtimes from doing something different. So coordination would be good.

Reviewer 3: @jasnell can you review please?
```suggestion
  * James – From WinterCG viewpoint, runtimes are well within their rights to add anything they want to `import.meta`, but maybe we should treat things in global scope which `import.meta` is. Should be treated as SemVer major.  The WinterCG already plans to add `import.meta.url`, `import.meta.resolve`, and `import.meta.main` to the minimum common API. That said, ok to add the bit of metadata being proposed, but ask would be some write up to WinterCG to get input to avoid other runtimes from doing something different. So coordination would be good.
```

  * Michael 2 concerns, would we add, do we get on WinterCG feedback.

>   * Antoine - Bun has already exposed with different names - Bun has installed the same fields with different names: <https://bun.sh/docs/api/import-meta>

Reviewer 3: ```suggestion
  * Antoine - Bun has installed the same fields with different names: <https://bun.sh/docs/api/import-meta>
```
